### PR TITLE
feat: connect displayText to storage browser interface

### PR DESCRIPTION
**Pull request by dindjarinjs**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/createStorageBrowser.tsx [hunk 1]
```diff
@@ -116,27 +118,27 @@ export function createStorageBrowser(input: CreateStorageBrowserInput): {
...
 
-  const StorageBrowser: StorageBrowserComponent = ({ views }) => (
+  const StorageBrowser: StorageBrowserComponent = ({ views, displayText }) => (
     <ErrorBoundary>
-      <Provider>
-        <ViewsProvider views={views}>
-          <StorageBrowserDefault />
-        </ViewsProvider>
-      </Provider>
+      <DisplayTextProvider displayText={displayText}>
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/createStorageBrowser.tsx:135]: Why is the change necessary?

File: packages/react-storage/src/components/StorageBrowser/composables/DataTable/headers/SortHeader.tsx [hunk 2]
```diff
@@ -23,7 +21,7 @@ export const SortHeader = ({ content }: SortHeaderProps): React.JSX.Element => {
       className={`${STORAGE_BROWSER_BLOCK_TO_BE_UPDATED}__table-sort-header`}
       onClick={onSort}
     >
-      {label && capitalize(label)}
```
Comment (ashika112) [inline packages/react-storage/src/components/StorageBrowser/composables/DataTable/headers/SortHeader.tsx:26]: why are we removing capitalization?

**Review by ashika112 — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/composables/DataTable/headers/SortHeader.tsx (hunk 2 above)
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/composables/DataTable/headers/SortHeader.tsx:26]: Capitalization is handled by the display text API

File: packages/react-storage/src/components/StorageBrowser/displayText/context.tsx [hunk 3]
```diff
@@ -15,6 +15,44 @@ export const { DisplayTextContext, useDisplayText } = createContextUtilities<
   errorMessage: '`useDisplayText` must be called inside `DisplayTextProvider`',
 });
 
+export function resolveDisplayText(
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/displayText/context.tsx:18]: Can we add a small unit test for this?

**Review by dbanksdesign — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by ashika112 — APPROVED** [review]

**Pull request merged**